clients/localdir/client.go

```diff
@@ -34,12 +34,12 @@ import (
 )
 
 var (
-	_                clients.RepoClient = &localDirClient{}
+	_                clients.RepoClient = &LocalDirClient{}
 	errInputRepoType                    = errors.New("input repo should be of type repoLocal")
 )
 
 //nolint:govet
-type localDirClient struct {
+type LocalDirClient struct {
```

Comment:
nit: now that we're exporting this type, lets make the name match the others:
```go
type Client struct {
```

so that users will write:
```go
case *localdir.Client:
```

instead of having the repetition
```go
case *localdir.LocalDirClient:
```